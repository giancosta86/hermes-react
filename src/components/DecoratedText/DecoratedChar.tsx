import React from "react";
import { CharMetadata } from "@/Metadata";

export interface DecoratedCharProps {
  char: string;
  metadata: CharMetadata;
}

export const DecoratedChar = ({ char, metadata }: DecoratedCharProps) =>
  metadata.annotation ? (
    <ruby className={metadata.className}>
      {char}
      <rp>(</rp>
      <rt>{metadata.annotation}</rt>
      <rp>)</rp>
    </ruby>
  ) : (
    <span className={metadata.className}>{char}</span>
  );
