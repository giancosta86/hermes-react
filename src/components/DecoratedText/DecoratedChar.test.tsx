import React from "react";
import { create } from "react-test-renderer";
import classnames from "classnames";
import { DecoratedChar } from "./DecoratedChar";

describe("<DecoratedChar>", () => {
  describe("when receiving a character with annotation and classes", () => {
    it("should render a <ruby> tag having classes", () => {
      const renderer = create(
        <DecoratedChar
          char="A"
          metadata={{
            annotation: "shū",
            className: classnames("alpha", "beta")
          }}
        />
      );

      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });

  describe("when receiving a character with annotation only", () => {
    it("should render a classless <ruby> tag", () => {
      const renderer = create(
        <DecoratedChar char="A" metadata={{ annotation: "shū" }} />
      );

      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });

  describe("when receving a character with classes and no annotation", () => {
    it("should render a <span> with classes", () => {
      const renderer = create(
        <DecoratedChar
          char="A"
          metadata={{ className: classnames("alpha", "beta") }}
        />
      );

      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
});
