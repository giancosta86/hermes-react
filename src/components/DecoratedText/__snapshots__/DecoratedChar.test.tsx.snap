// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DecoratedChar> when receiving a character with annotation and classes should render a <ruby> tag having classes 1`] = `
<ruby
  className="alpha beta"
>
  A
  <rp>
    (
  </rp>
  <rt>
    shū
  </rt>
  <rp>
    )
  </rp>
</ruby>
`;

exports[`<DecoratedChar> when receiving a character with annotation only should render a classless <ruby> tag 1`] = `
<ruby>
  A
  <rp>
    (
  </rp>
  <rt>
    shū
  </rt>
  <rp>
    )
  </rp>
</ruby>
`;

exports[`<DecoratedChar> when receving a character with classes and no annotation should render a <span> with classes 1`] = `
<span
  className="alpha beta"
>
  A
</span>
`;
